iree/hal/metal/BUILD

```diff
@@ -0,0 +1,203 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Default implementations for HAL types that use the host resources.
+# These are generally just wrappers around host heap memory and host threads.
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+objc_library(
+    name = "metal_capture_manager",
+    hdrs = ["metal_capture_manager.h"],
+    copts = ["-std=c++14"],
+    non_arc_srcs = ["metal_capture_manager.mm"],
+    sdk_frameworks = ["Metal"],
```

Comment:
I _think_ these rules are unified enough between open source Bazel and what we use internally, though I recall that not always being true...